crates/bevy_picking/src/hover.rs

```diff
@@ -275,3 +275,266 @@ fn merge_interaction_states(
         new_interaction_state.insert(*hovered_entity, new_interaction);
     }
 }
+
+/// A component that allows users to use regular Bevy change detection to determine when the pointer
+/// enters or leaves an entity. Users should insert this component on an entity to indicate interest
+/// in knowing about hover state changes.
+///
+/// This is similar to the old Bevy [`Interaction`] component, except that it only tracks
+/// hover state, not button presses or other interactions.
+///
+/// The component's boolean value will be `true` whenever the pointer is currently hovering over the
+/// entity, or any of the entity's children. This is consistent with the behavior of the CSS
+/// `:hover` pseudo-class, which applies to the element and all of its children.
```

Comment:
I think it makes sense to say "descendants" here instead of "children" as this applies to arbitrary levels of the hierarchy. It would also be good to use "directly" here, to connect this to HoveredDirect.

```suggestion
/// The component's boolean value will be `true` whenever the pointer is currently directly hovering over the
/// entity, or any of the entity's descendants (as defined by the [`ChildOf`] relationship). This is consistent with the behavior of the CSS
/// `:hover` pseudo-class, which applies to the element and all of its children.
```